### ISSUE TITLE
Error in the DDP mode

### ISSUE THREAD

**Issue by jinxinglu** [joanaapa/Distillation-DNABERT-Promoter#1]
when using the DDP mode to train the model, it would raise the error of "This error indicates that your module has parameters that were not used in producing loss".
Since the minilm model only uses the attention parameters, so the parameters of student model like "bert.encoder.layer.-1.output.xx" and "cls.predictions.transform.xx" would have no gradient updates.  So how to fix this problem? thanks.

**Comment by joanaapa** [comment]
You should be able to circumvent the issue by adding the find_unused_parameters option in DDP ([see documentation](https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html)), so make sure it is set to True. Otherwise you can just remove the last layers from MiniLM during distillation.

I didn't use DDP with MiniLM so I was not aware of this problem. Thanks for letting me know. Hope this helps!

**Comment by jinxinglu** [comment]
Thanks for responding quickly. I have set the find_unused_parameters=True when using the DDP mode. I'm trying to find the solution to this problem.
btw, I have confused about MiniLM. During distillation, some parameters of the last layer of the student model (4-layer), e.g. "bert.encoder.layer.3.output.LayerNorm", "bert.encoder.layer.3.output.dense", "bert.encoder.layer.3.intermediate" , would be always not updated because they are not used in the calculation of loss (loss_at+loss_vr). So these parameters would keep the value when the model is initialized? Is it right?

**Comment by joanaapa** [comment]
Yes exactly

**Comment by jinxinglu** [comment]
I change the codes of loss function 
from `loss = loss_at + loss_vr` to `loss = loss_at + loss_vr + 1e-8 * loss_mlm`
then DDP works well

**Comment by joanaapa** [comment]
Thanks for sharing the solution!
